### PR TITLE
changes to blockchain k plugin for KEVM

### PR DESCRIPTION
**Pull request by dwightguth**

### CODE REVIEW

File: vm/VM.ml [hunk 1]
```diff
@@ -120,13 +120,13 @@ let pack_output rets =
...
 | Iele_config -> [KApply0(parse_klabel "ALBE_IELE-GAS")]
 | Ethereum_config cfg ->
-  let homestead = World.to_z_unsigned cfg.homestead_block_number in
-  if Z.lt blocknum homestead then [KApply0(parse_klabel "FRONTIER_EVM")] else
-  let eip150 = World.to_z_unsigned cfg.eip150_block_number in
-  if Z.lt blocknum eip150 then [KApply0(parse_klabel "HOMESTEAD_EVM")] else
-  let eip161 = World.to_z_unsigned cfg.eip161_block_number in
-  if Z.lt blocknum eip161 then [KApply0(parse_klabel "EIP150_EVM")] else
-  [KApply0(parse_klabel "EIP158_EVM")]
+  let eip161 = World.to_z cfg.eip161_block_number in
```
Comment (bmmoore) [inline vm/VM.ml]: Why did this change from `to_z_unsigned` to `to_z`?

**Comment by bmmoore** [comment]
I don't know why that comment is marked "outdated", it's asking why `to_z_unsigned` changed to `to_z`. Was that guaranteed by the protocol all along?

**Comment by dwightguth** [comment]
On the mantis side, the block numbers at which certain hard forks become active can be negative to indicate that that hard fork does not exist on that chain. That's why you see this.

**Comment by bmmoore** [comment]
These changes look ok too.

**Pull request merged**